plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/delete/TrinoDeleteFile.java

```diff
@@ -0,0 +1,339 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.delete;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.trino.plugin.iceberg.IcebergSplit;
+import org.apache.iceberg.DeleteFile;
+import org.apache.iceberg.FileContent;
+import org.apache.iceberg.FileFormat;
+import org.apache.iceberg.StructLike;
+import org.openjdk.jol.info.ClassLayout;
+
+import javax.annotation.Nullable;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+import java.util.function.ToLongFunction;
+
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static io.airlift.slice.SizeOf.estimatedSizeOf;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A Jackson-serializable implementation of the Iceberg {@link DeleteFile}. Allows for delete files to be included in {@link IcebergSplit}.
+ */
+public class TrinoDeleteFile
+        implements DeleteFile
+{
+    public static final long INSTANCE_SIZE = ClassLayout.parseClass(TrinoDeleteFile.class).instanceSize();
+
+    @Nullable private final Long pos;
+    private final int specId;
+    private final FileContent fileContent;
+    private final String path;
+    private final FileFormat format;
+    private final long recordCount;
+    private final long fileSizeInBytes;
+    @Nullable private final Map<Integer, Long> columnSizes;
+    @Nullable private final Map<Integer, Long> valueCounts;
+    @Nullable private final Map<Integer, Long> nullValueCounts;
+    @Nullable private final Map<Integer, Long> nanValueCounts;
+    @Nullable private final Map<Integer, byte[]> lowerBounds;
+    @Nullable private final Map<Integer, byte[]> upperBounds;
```

Comment:
would it make the code simpler if we had empty maps instead of nulls?

the `copyOf` would still need to handle the flexibility of the `DeleteFile` (having `firstNonNull(..., ImmutableMap.of())` would be sufficient for that, without `?:`),
and the rest of the class would be simpler

or, is it a semantical difference between an empty map and a null?

Reply:
>  is it a semantical difference between an empty map and a null?

There isn't in our code but the Iceberg API treats the Maps as nullable and I don't know if there's a distinction in how it's used there so I'd rather leave it.